homework/12_V/15_Yordan/Homework1/get_intersections.py

```diff
@@ -0,0 +1,23 @@
+import math
+
+def get_intersections(x0, y0, r0, x1, y1, r1):
+
+    d = math.sqrt((x1 - x0) ** 2 + (y1 - y0) ** 2)
+    if d > r0 + r1:
+        return None
+    if d < abs(r0 - r1):
+        return None
+    if d == 0 and r0 == r1:
+        return None
+    else:
+        a = (r0 ** 2 - r1 ** 2 + d ** 2) / (2 * d)
+        h = math.sqrt(r0 ** 2 - a ** 2)
+        x2 = x0 + a * (x1 - x0) / d
+        y2 = y0 + a * (y1 - y0) / d
+        x3 = x2 + h * (y1 - y0) / d
+        y3 = y2 - h * (x1 - x0) / d
+
+        x4 = x2 - h * (y1 - y0) / d
+        y4 = y2 + h * (x1 - x0) / d
+
+        return (x3, y3, x4, y4)
```

Comment:
It appears you have solved a slightly different task than the one I had in mind. You calculate the points of intersection while the task was to determine if and how two circles intersect - if they intersect in 2 points, if they touch in a single point, if one contains the other or if they don't intersect at all. For that reason I will give you a 5.

Reply:
Scratch that. You also get an A for effort.